From 88d42e41a028d8f311590cc93fa6f7c1eccd2520 Mon Sep 17 00:00:00 2001
From: Mohammed Sameer <msameer@foolab.org>
Date: Mon, 14 Apr 2014 22:23:11 +0300
Subject: [PATCH 1/5] Set video branch to NULL after finishing video recording.

Our encoders seem to cause a kernel oops and a reboot when we start
recording for the second time.
The only workaround for now is to set them to NULL state and back to PLAYING
---
 gst/camerabin2/gstcamerabin2.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/camerabin2/gstcamerabin2.c b/gst/camerabin2/gstcamerabin2.c
index c77585a..c101637 100644
--- a/gst/camerabin2/gstcamerabin2.c
+++ b/gst/camerabin2/gstcamerabin2.c
@@ -953,10 +953,10 @@ gst_camera_bin_video_reset_elements (gpointer u_data)
   gst_camera_bin_finish_video_file (camerabin);
 
   /* reset element states to clear eos/flushing pads */
-  gst_element_set_state (camerabin->video_encodebin, GST_STATE_READY);
-  gst_element_set_state (camerabin->videobin_capsfilter, GST_STATE_READY);
+  gst_element_set_state (camerabin->video_encodebin, GST_STATE_NULL);
+  gst_element_set_state (camerabin->videobin_capsfilter, GST_STATE_NULL);
   if (camerabin->video_filter) {
-    gst_element_set_state (camerabin->video_filter, GST_STATE_READY);
+    gst_element_set_state (camerabin->video_filter, GST_STATE_NULL);
     gst_element_sync_state_with_parent (camerabin->video_filter);
   }
   gst_element_sync_state_with_parent (camerabin->videobin_capsfilter);
-- 
1.8.4.rc3

